### ISSUE TITLE
MAINT,DOC: Classic Flang is not LLVM Flang

### ISSUE THREAD

**Issue by HaoZeke** [numpy/numpy#20240]
### Issue with current documentation:

https://github.com/numpy/numpy/commit/3ff922786efcfbee29ae2636be8b381f5e376f69 introduced classic flang based on `pgifortran` into `np.distutils`.

### Idea or request for content:

This is both unmaintained and confusing for users or to-be users of LLVM Flang. Removal would be for the best.

**Comment by HaoZeke** [comment]
Also while we're at it we might as well kill off (in separate commits) the[ other unmaintained Fortran compilers](https://fortran-lang.org/compilers/)

**Comment by rgommers** [comment]
I don't think this is a good first issue. We cannot just remove this. Right now Classic Flang still has more functionality I believe, in particular on Windows. 

We may add LLVM-FLang as a new compiler and improve the docs.

**Comment by rgommers** [comment]
Also, Classic Flang is not unmaintained. As of right now it's my best hope at a sane Fortran story for SciPy within the next year.

**Comment by HaoZeke** [comment]
Ah, I think I made a bit of a knee jerk assumption based off [their wiki](https://github.com/flang-compiler/flang/wiki) which states that it should be considered to be superseded by F18 from early 2020.

However, the impetus was that at-least in the context of `clang` and `flang` optimizations as discussed in #19930, classic `flang` is incorrect.

Naming the fortran compilers is a bit of a nightmare, perhaps `classic_flang` and `flang` for LLVM would make more sense for the issue?

**Comment by rgommers** [comment]
> Naming the fortran compilers is a bit of a nightmare, perhaps `classic_flang` and `flang` for LLVM would make more sense for the issue?

I think I'd prefer a non-breaking change. Can we just add some code comments for now that `flang` means classic, and name the new one `flang_llvm`? Or maybe rename the other `classic_flang` but keep `flang` as an alias to that one (I think that is possible)? Once the new one actually is the better compiler, we can then switch `flang`. My impression is that that's a couple of years away still.

**Comment by HaoZeke** [comment]
> Or maybe rename the other classic_flang but keep flang as an alias to that one (I think that is possible)? 
This sounds like a plan. Will do.
